### PR TITLE
Patch package and deploy lab

### PR DESCRIPTION
**Pull request by rezabekf**
*Issue #, if available:*
N/A
*Description of changes:*
This PR adds all the code to solutions folder to maintain consistency.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# From code/package-and-deploy directory run ./scripts/invoke-lambda-function.sh
+
+LAMBDA_FUNCTION_NAME="cfn-workshop-python-function"
+
+PAYLOAD='{"time_zone":"Europe/London"}'
```
Comment (HieronymusLex) [inline code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh:7]: we should probably let this be user defined given it is a user specific settings, then provide a default

File: code/solutions/package-and-deploy/infrastructure.template [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+            Condition: {}
+      Path: /
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - arn:aws:iam::aws:policy/AWSXrayFullAccess
```
Comment (HieronymusLex) [inline code/solutions/package-and-deploy/infrastructure.template:19]: is x-ray full access really required?

File: code/solutions/package-and-deploy/lambda/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+pytz==2019.3
```
Comment (HieronymusLex) [inline code/solutions/package-and-deploy/lambda/requirements.txt:1]: this is quite an old version, can we upgrade?

File: code/solutions/package-and-deploy/scripts/deploy-cfn-template.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# From code/package-and-deploy directory run ./scripts/deploy-cfn-template.sh
+
+BUCKET_NAME="cfn-workshop-s3-s3bucket-1wo3wjmf93viq" # use your own uniquely named S3 bucket
```
Comment (HieronymusLex) [inline code/solutions/package-and-deploy/scripts/deploy-cfn-template.sh:5]: if the user has to use their own s3 bucket, we should be requiring that as input to the script and erroring if not provided. same goes for region

File: code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+aws lambda invoke \
+  --function-name $LAMBDA_FUNCTION_NAME \
+  --payload $PAYLOAD \
+  response.json
+
+echo "$(<response.json)"
```
Comment (HieronymusLex) [inline code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh:14]: cat?

Comment (HieronymusLex) [reply code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh:14]: or use stdout as the file for the preceding command?

**Review by HieronymusLex — CHANGES_REQUESTED** [review]

File: code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh (hunk 1 above)
Comment (rezabekf) [reply code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh]: Done

File: code/solutions/package-and-deploy/infrastructure.template (hunk 2 above)
Comment (rezabekf) [reply code/solutions/package-and-deploy/infrastructure.template]: Removed

File: code/solutions/package-and-deploy/lambda/requirements.txt (hunk 3 above)
Comment (rezabekf) [reply code/solutions/package-and-deploy/lambda/requirements.txt]: Upgraded

File: code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh (hunk 5 above)
Comment (rezabekf) [reply code/solutions/package-and-deploy/scripts/invoke-lambda-function.sh]: Updated

File: code/workspace/package-and-deploy/scripts/deploy-cfn-template.sh [hunk 6]
```diff
@@ -2,28 +2,27 @@
 
 # From code/package-and-deploy directory run ./scripts/deploy-cfn-template.sh
 
-BUCKET_NAME="cfn-workshop-s3-s3bucket-1wo3wjmf93viq" # use your own uniquely named S3 bucket
-PREFIX_NAME="lambda-function"
-STACK_NAME="cfn-workshop-lambda"
+BUCKET_NAME="rezabekf-development-bucket" # cfn-workshop-s3bucket-unique-name # use your own uniquely named S3 bucket
```
Comment (HieronymusLex) [inline code/workspace/package-and-deploy/scripts/deploy-cfn-template.sh:5]: same thing here with requiring this as an arg

Comment (rezabekf) [reply code/workspace/package-and-deploy/scripts/deploy-cfn-template.sh]: Sorted

**Review by HieronymusLex — APPROVED** [review]

**Pull request merged**